src/popover/popover-animation-from-top.jsx

```diff
@@ -58,56 +79,24 @@ const PopoverAnimationFromTop = React.createClass({
   },
 
   componentWillReceiveProps(nextProps, nextContext) {
-    let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
-
     this.setState({
       open: nextProps.open,
-      muiTheme: newMuiTheme,
+      muiTheme: nextContext.muiTheme || this.state.muiTheme,
     });
   },
 
-  getStyles() {
-    let {targetOrigin} = this.props;
-    let horizontal = targetOrigin.horizontal.replace('middle', 'vertical');
-
-    return {
-      base: {
-        opacity: 0,
-        transform: 'scaleY(0)',
-        transformOrigin: `${horizontal} ${targetOrigin.vertical}`,
-        position: 'fixed',
-        zIndex: this.state.muiTheme.zIndex.popover,
-        transition: Transitions.easeOut('450ms', ['transform', 'opacity']),
-        maxHeight: '100%',
-      },
-
-    };
-  },
-
-  getOpenStyles() {
-    return {
-      base: {
-        opacity: 1,
-        transform: 'scaleY(1)',
-      },
-    };
-  },
-
   render() {
-    let {
+    const {
       className,
       style,
       zDepth,
     } = this.props;
 
-    let styles = this.getStyles();
-    let openStyles = {};
-    if (this.state.open)
-      openStyles = this.getOpenStyles();
+    const styles = getStyles(this.props, this.state);
 
     return (
       <Paper
-        style={this.mergeStyles(styles.base, style, openStyles.base)}
```

Comment:
I take it that the reason why `styles.base` is separate from `openStyles.base` is so that developers can't override `openStyles`.  I really wonder if that additional logic is worth it though for what we're getting for it. Right now `position` is overridable and that would completely break this component. In addition `transformOrigin` isn't being protected either.

Personally, I think it's easier on us and the developer not make any exceptions about what they can and can't override. I think the "protection mechanism" is too brittle for it to be worth it in this case. I'm open to changing it back though!